csvimport_connector.py

```diff
@@ -0,0 +1,358 @@
+
+        ret_val, response = self._make_rest_call('/rest/artifact?_filter_container_id={0}&page_size={1}'.format(
+            container_id, page_size), action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+        fieldnames = set()
+        for i in response['data']:
+            for key, val in list(i.items()):
+                if isinstance(val, dict) and key != 'cef_types':
+                    temp = i.pop(key)
+                    i.update(temp)
+            for j in list(i.keys()):
+                fieldnames.add(j)
+        curr_time = datetime.now().strftime("%Y_%m_%d_%I_%M_%S")
+        filename = "csv_output_" + curr_time + ".csv"
+        file_loc = "/opt/phantom/vault/tmp/" + filename
+        with open(file_loc, 'w') as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=list(fieldnames))
+            writer.writeheader()
+            for row in response['data']:
+                writer.writerow(row)
+        vault_ret_dict = phantomrules.vault_add(container=container_id,
+                                                file_location=file_loc,
+                                                file_name=filename)
+        if vault_ret_dict[0]:
+            vault_details = {
+                phantom.APP_JSON_VAULT_ID: vault_ret_dict[2],
+                'file_name': filename
+            }
+            action_result.add_data(vault_details)
+            return action_result.set_status(phantom.APP_SUCCESS, "Successfully Created CSV")
+
+        return action_result.set_status(phantom.APP_ERROR, 'Error adding file to vault: {0}'.format(vault_ret_dict))
+
+    def _handle_ingest_csv(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        ret_val, container_id = self._validate_integer(action_result, param["container_id"], 'Container ID', False)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+        vault_id = param['vault_id']
+        artifact_name = param['artifact_name']
+        artifact_label = param.get('artifact_label', 'events')
+        cef_names = param['cef_column_headers']
+
+        if cef_names:
+            cef_names = [x.strip() for x in cef_names.split(',')]
+            cef_names = list(filter(None, cef_names))
+        num_columns = len(cef_names)
+
+        # Make sure container exists first, provide a better error message than waiting for save_artifacts to fail
+        ret_val, message, _ = self.get_container_info(container_id)
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(phantom.APP_ERROR, "Unable to find container: {}".format(message))
+
+        # run_automation = param.get('run_automation', True)
+        if vault_id:
+            ret_val, file_info = self._get_file_info_from_vault(action_result, vault_id)
+            if phantom.is_fail(ret_val):
+                return ret_val
+            self.debug_print("File Info", file_info)
```

Comment:
Why not use File ID or Vault ID?

Reply:
@alexa-phantom inside file_info we have both the information about file_ID as well Vault_ID